server/src/main/scala/org/http4s/server/middleware/CORS.scala

```diff
@@ -41,13 +41,13 @@ object CORS {
    * Currently, you cannot make permissions depend on request details
    */
   def apply(service: HttpService, config: CORSConfig = DefaultCORSConfig): HttpService = Service.lift { req =>
-    
+
     // In the case of an options request we want to return a simple response with the correct Headers set.
     def createOptionsResponse(origin: Header, acrm: Header): Response = corsHeaders(origin.value, acrm.value)(Response())
 
     def corsHeaders(origin: String, acrm: String)(resp: Response): Response =
       config.allowedHeaders.map(_.mkString("", ", ", "")).fold(resp) { hs =>
-        resp.putHeaders(Header("Access-Control-Allow-Headers", hs))
+        resp.putHeaders(Header("Access-Control-Allow-Headers", hs), Header("Access-Control-Expose-Headers", hs))
```

Comment:
I think `Access-Control-Allow-Headers` is supposed to go on the preflight request, whereas `Access-Control-Expose-Headers` is supposed to go on the "simple cross origin request" or "actual request"

https://www.w3.org/TR/cors/#simple-cross-origin-request

I think maybe `corsHeaders` is being shared in two cases when it needs to be split.  Am I reading the spec right?